simulation/behavior_tree_plugin/include/behavior_tree_plugin/vehicle/follow_trajectory_sequence/polyline_trajectory_follower.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef BEHAVIOR_TREE_PLUGIN__VEHICLE__TRAJECTORY_FOLLOWER_HPP_
+#define BEHAVIOR_TREE_PLUGIN__VEHICLE__TRAJECTORY_FOLLOWER_HPP_
+
+#include <behavior_tree_plugin/vehicle/vehicle_action_node.hpp>
+#include <optional>
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/data_type/follow_trajectory.hpp>
+
+namespace entity_behavior
+{
+namespace vehicle
+{
+
+class PolylineTrajectoryFollower
+{
+public:
+  PolylineTrajectoryFollower() = default;
+  ~PolylineTrajectoryFollower() = default;
+
+  using PolylineTrajectoryParameter_ptr =
+    std::shared_ptr<traffic_simulator::follow_trajectory::Parameter<
+      traffic_simulator::follow_trajectory::Polyline>>;
+
+  PolylineTrajectoryParameter_ptr trajectory_parameter_m;
+  traffic_simulator_msgs::msg::EntityStatus entity_status_m;
+  traffic_simulator_msgs::msg::BehaviorParameter behavior_parameter_m;
+  std::optional<traffic_simulator_msgs::msg::VehicleParameters> vehicle_parameters_m;
+  double step_time_m;
+
+  std::optional<traffic_simulator_msgs::msg::EntityStatus> followTrajectory(
+    const traffic_simulator_msgs::msg::EntityStatus &, PolylineTrajectoryParameter_ptr &,
```

Comment:
to EntityStatus tez nie ulega zmianie? ciekawe

Reply:
nie zmienia się - `followTrajectory` zwraca updated_status, który potem w `FollowPolylineTrajectoryAction::tick` jest ustawiany przez `setOutput`. Tak samo się działo przy `makeUpdatedStatus`